<h1>Chapter 2 | Data Exercise #1 | <code>hotels-europe</code>: Data Preparation</h1>

<p>1. Use the <code>hotels-europe</code> data for another city (not Vienna).</p>
<p>Assignments</p>
<ul>
    <li>Load and clean the data.</li>
    <li>Document the cleaning.</li>
    <li>Describe the clean dataset.</li>
    <li>Look at the raw data and make it into tidy data.</li>
</ul>

<h2><b>1.</b> Load the data</h2>

In [50]:
import os
import sys
import warnings
import pandas as pd

warnings.filterwarnings("ignore")

In [51]:
# Set pandas options to display 500 rows as max
pd.set_option("display.max_rows", None)

In [52]:
# Current script folder
current_path = os.getcwd()
dirname = current_path.split("da_data_exercises")[0]

# Get location folders
data_in = f"{dirname}da_data_repo/hotels-europe/"
data_out = f"{dirname}da_data_exercises/ch02-preparing_data_for_analysis/data/"
output = f"{dirname}da_data_exercises/ch02-preparing_data_for_analysis/data/output/"
func = f"{dirname}da_case_studies/ch00-tech_prep/"
sys.path.append(func)


In [53]:
# import the prewrriten helper functions
from py_helper_functions import *

<p>Ok. We can emulate the example on <code>hotels-vienna</code> to see if we can get the same result with any particular city in Europe. First, let's get hold of the raw dataset, which contains all hotel ratings in Europe. We will use <b>London</b> as our target city for this exercise.</p>

In [54]:
# Set raw data working directory
data_in_raw = f"{data_in}raw/"

In [55]:
data = pd.read_csv(f"{data_in_raw}hotelbookingdata.csv")
data.head()

addresscountryname city_actual  rating_reviewcount center1distance  \
0        Netherlands   Amsterdam              1030.0       3.1 miles   
1        Netherlands   Amsterdam              1030.0       3.1 miles   
2        Netherlands   Amsterdam              1030.0       3.1 miles   
3        Netherlands   Amsterdam              1030.0       3.1 miles   
4        Netherlands   Amsterdam              1030.0       3.1 miles   

  center1label center2distance      center2label neighbourhood  price  \
0  City centre       3.6 miles  Montelbaanstoren     Amsterdam    172   
1  City centre       3.6 miles  Montelbaanstoren     Amsterdam    122   
2  City centre       3.6 miles  Montelbaanstoren     Amsterdam    122   
3  City centre       3.6 miles  Montelbaanstoren     Amsterdam    552   
4  City centre       3.6 miles  Montelbaanstoren     Amsterdam    122   

          price_night  ...  accommodationtype  guestreviewsrating  \
0   price for 1 night  ...  _ACCOM_TYPE@Hotel              4.3 /5   
1   price for 1 night  ...  _ACCOM_TYPE@Hotel              4.3 /5   
2   price for 1 night  ...  _ACCOM_TYPE@Hotel              4.3 /5   
3  price for 4 nights  ...  _ACCOM_TYPE@Hotel              4.3 /5   
4   price for 1 night  ...  _ACCOM_TYPE@Hotel              4.3 /5   

   scarce_room  hotel_id offer     offer_cat  year  month  weekend holiday  
0            0       1.0     0   0% no offer  2017     11        1       0  
1            0       1.0     1  15-50% offer  2018      1        1       0  
2            0       1.0     1  15-50% offer  2017     12        0       1  
3            0       1.0     1   1-15% offer  2017     12        0       1  
4            0       1.0     1  15-50% offer  2018      2        1       0  

[5 rows x 24 columns]

<p>Now, our first step is to check the cities available in the dataset. We can get a list of them using <code>.unique()</code>:

In [56]:
data["city_actual"].unique()

array(['Amsterdam', 'Badhoevedorp', 'Diemen', 'Halfweg', 'Hoofddorp',
       'Lijnden', 'Duivendrecht', 'Schiphol', 'Schiphol-Rijk',
       'Zwanenburg', 'Acharnes', 'Athens', 'Agia Paraskevi', 'Alimos',
       'Glyfada', 'Vari-Voula-Vouliagmeni', 'Elliniko-Argyroupoli',
       'Palaio Faliro', 'Saronikos', 'Lavreotiki', 'Chaidari',
       'Chalandri', 'Dionysos', 'Elefsina', 'Kallithea', 'Kifisia',
       'Lykovrysi-Pefki', 'Mandra-Eidyllia', 'Marathon',
       'Markopoulo Mesogaias', 'Marousi', 'Megara', 'Metamorfosi',
       'Moschato-Tavros', 'Nea Smirni', 'Oropos', 'Pallini', 'Piraeus',
       'Rafina-Pikermi', 'Salamis', 'Spata-Artemida', 'Barcelona',
       'Castelldefels', 'Cornella de Llobregat', 'El Prat de Llobregat',
       'Esplugues de Llobregat', 'Gava', 'barcelona',
       'L\\u0027Hospitalet de Llobregat', 'Molins de Rei',
       'Sant Adria de Besos', 'Sant Joan Despi', 'Sant Just Desvern',
       'Belgrade', 'Berlin', 'Schoenefeld', 'Birmingham', 'Walsall',
       'B

<p>Ok. Just to make sure, let's use <code>.unique()</code> together with <code>in</code> to check if there are any observations with "London" in the variable <code>"city_actual"</code>

In [57]:
"London" in data["city_actual"].unique()

True

<p>Great. Now, we can filter our dataset and keep only observations for acommodations in London:</p>

In [58]:
df = data.loc[data["city_actual"] == "London"]
df.head()

addresscountryname city_actual  rating_reviewcount center1distance  \
52159     United Kingdom      London               161.0       6.5 miles   
52160     United Kingdom      London               161.0       6.5 miles   
52161     United Kingdom      London               161.0       6.5 miles   
52162     United Kingdom      London               161.0       6.5 miles   
52163     United Kingdom      London               161.0       6.5 miles   

      center1label center2distance                        center2label  \
52159  City centre       9.3 miles  Wimbledon Park Underground Station   
52160  City centre       9.3 miles  Wimbledon Park Underground Station   
52161  City centre       9.3 miles  Wimbledon Park Underground Station   
52162  City centre       9.3 miles  Wimbledon Park Underground Station   
52163  City centre       9.3 miles  Wimbledon Park Underground Station   

      neighbourhood  price         price_night  ...  accommodationtype  \
52159    Blackheath    515  price for 4 nights  ...  _ACCOM_TYPE@Hotel   
52160    Blackheath    147   price for 1 night  ...  _ACCOM_TYPE@Hotel   
52161    Blackheath    124   price for 1 night  ...  _ACCOM_TYPE@Hotel   
52162    Blackheath    135   price for 1 night  ...  _ACCOM_TYPE@Hotel   
52163    Blackheath    128   price for 1 night  ...  _ACCOM_TYPE@Hotel   

       guestreviewsrating  scarce_room  hotel_id offer    offer_cat  year  \
52159                4 /5            0    7164.0     0  0% no offer  2017   
52160                4 /5            0    7164.0     0  0% no offer  2018   
52161                4 /5            0    7164.0     0  0% no offer  2018   
52162                4 /5            0    7164.0     0  0% no offer  2018   
52163                4 /5            0    7164.0     0  0% no offer  2018   

       month  weekend holiday  
52159     12        0       1  
52160      6        1       0  
52161      3        1       0  
52162      4        1       0  
52163      2        1       0  

[5 rows x 24 columns]

<p>Let's start by tackling the variables <code>center1distance</code> and <code>center2distance</code>. Although they represent the distance to center from each hotel, both were stored as strings/objects, with one decimal. We can transform them to numerical so that they can be properly used during our analysis.</p>

In [59]:
df["distance"] = df["center1distance"].str.split(" ").apply(lambda x: float(x[0]))
df["distance_alter"] = df["center2distance"].str.split(" ").apply(lambda x: float(x[0]))

<p>In the above codes, we applied <code>.str.split()</code> to split the observations using a whitespace, as values were stored as "n miles". Then, we applied <code>lambda</code>, which converts each initial value of the split value to a float.</p>

In [60]:
df[["distance", "distance_alter"]]

distance  distance_alter
52159       6.5             9.3
52160       6.5             9.3
52161       6.5             9.3
52162       6.5             9.3
52163       6.5             9.3
52164       6.5             9.3
52165       6.5             9.3
52166       6.5             9.3
52167       6.5             9.3
52168       6.5             9.3
52169       6.7             9.9
52170       6.7             9.9
52171       6.7             9.9
52172       6.7             9.9
52173       6.7             9.9
52174       6.7             9.9
52175       6.7             9.9
52176       6.7             9.9
52177       6.7             9.9
52178       6.7             9.9
52179       1.1             6.8
52180       1.3             7.1
52181       1.3             7.1
52182       1.3             7.1
52183       1.3             7.1
52184       1.0             6.9
52185       1.0             6.9
52186       1.0             6.9
52187       1.0             6.9
52188       1.0             6.9
52189       0.8             6.5
52190       0.8             6.5
52191       0.8             6.5
52192       0.8             6.5
52193       0.8             6.5
52194       0.8             6.5
52195       0.8             6.5
52196       0.8             6.5
52197       0.8             6.5
52198       0.7             6.5
52199       0.7             6.5
52200       0.7             6.5
52201       0.7             6.5
52202       0.7             6.5
52203       0.7             6.5
52204       0.7             6.5
52205       0.7             6.5
52206       0.9             6.8
52207       0.9             6.8
52208       0.9             6.8
52209       0.9             6.8
52210       0.9             6.8
52211       0.9             6.8
52212       0.9             6.8
52213       0.9             6.8
52214       0.9             6.8
52215       0.9             6.8
52216       0.9             6.6
52217       0.9             6.6
52218       0.9             6.6
52219       0.9             6.6
52220       0.9             6.6
52221       0.9             6.6
52222       0.9             6.6
52223       0.9             6.6
52224       0.9             6.8
52225       0.9             6.8
52226       0.9             6.8
52227       0.9             6.8
52228       0.9             6.8
52229       0.9             6.8
52230       0.9             6.8
52231       0.9             6.8
52232       0.9             6.8
52233       0.9             6.7
52234       0.9             6.7
52235       0.9             6.7
52236       0.9             6.7
52237       0.9             6.7
52238       0.9             6.7
52239       0.9             6.7
52240       0.9             6.7
52241       0.9             6.7
52242       0.9             6.7
52243       0.7             6.5
52244       0.7             6.5
52245       0.7             6.5
52246       0.7             6.5
52247       0.7             6.5
52248       0.7             6.5
52249       0.7             6.5
52250       0.7             6.5
52251       0.7             6.5
52252       1.1             6.7
52253       1.1             6.7
52254       1.1             6.7
52255       1.1             6.7
52256       1.1             6.7
52257       1.1             6.7
52258       1.1             6.7
52259       1.1             6.7
52260       1.1             6.7
52261       0.9             6.8
52262       1.2             7.0
52263       0.9             6.7
52264       0.9             6.7
52265       0.9             6.7
52266       0.9             6.7
52267       0.9             6.7
52268       0.9             6.7
52269       0.9             6.7
52270       1.0             6.9
52271       1.2             7.0
52272       1.2             7.0
52273       1.2             7.0
52274       0.9             6.6
52275       0.9             6.6
52276       0.9             6.6
52277       0.9             6.6
52278       0.9             6.6
52279       0.9             6.6
52280       0.9             6.6
52281       0.9             6.6
52282       0.9             6.6
52283  

In [61]:
df[["distance", "distance_alter"]].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8774 entries, 52159 to 61919
Data columns (total 2 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   distance        8774 non-null   float64
 1   distance_alter  8774 non-null   float64
dtypes: float64(2)
memory usage: 205.6 KB


<p>Great! Now, let's check <code>"accommodation_type</code>. All observations were stored with the prefix <code>_ACCOM_TYPE@</code>. Let's apply the procedure we've just used, with the difference that we will split values using <code>@</code> and getting hold of the value after it, and not before.</p>

In [62]:
df["accomodation_type"] = (
    df["accommodationtype"].str.split("@").apply(lambda x: x[1]).str.strip()
)

In [63]:
df["accomodation_type"]

52159                  Hotel
52160                  Hotel
52161                  Hotel
52162                  Hotel
52163                  Hotel
52164                  Hotel
52165                  Hotel
52166                  Hotel
52167                  Hotel
52168                  Hotel
52169            Guest House
52170            Guest House
52171            Guest House
52172            Guest House
52173            Guest House
52174            Guest House
52175            Guest House
52176            Guest House
52177            Guest House
52178            Guest House
52179              Apartment
52180      Bed and breakfast
52181      Bed and breakfast
52182      Bed and breakfast
52183      Bed and breakfast
52184                  Hotel
52185                  Hotel
52186                  Hotel
52187                  Hotel
52188                  Hotel
52189                  Hotel
52190                  Hotel
52191                  Hotel
52192                  Hotel
52193         

<p>We can now clean the variable <code>"price_night"</code>. We want to get hold of the number of nights that the price value refers to.</p>

In [64]:
df["nnight"] = df["price_night"].str.split(" ").apply(lambda x: int(x[2]))

<p>Here, we splitted the string using whitespace, which resulted in four substrings (ie "Price per 1 night). As we are intersted in the numeric value, we used <code>x[2]</code> to apply the lambda function.</p>
<p>Now, we can clean the variable <code>"guestreviewsrating</code>. The rating was stored as a value between 0 and 5 over its total, like 4.5/5. We just want the rating itself, hence let's split this variable like the others and convert the first value to a float.</p> 

In [65]:
df["rating"] = (
    df["guestreviewsrating"]
    .str.split(" ")
    .apply(lambda x: float(x[0]) if type(x) == list else None)
)

<p>I'll now use the authors' <code>tabulate</code> function, which returns the parameter values in a tabulated form containing the frequency of observations for each rating <code>Freq.</code>, the <code>Perc.</code> (percentage) of occurrencies for each rating, as well as a cummulative sum <code>Cum</code>.</p>

In [66]:
def tabulate(series, drop_missing=False):
    """Tabulate a pandas Series and return statistical observations."""
    table = (
        pd.concat(
            [
                series.value_counts(dropna=drop_missing)
                .sort_index()
                .round(2)
                .rename("Freq."),
                series.value_counts(normalize=True, dropna=drop_missing)
                .sort_index()
                .rename("Perc."),
            ],
            axis=1
        )
        .assign(Cum=lambda x: x["Perc."].cumsum())
        .round(3)
    )
    return table

In [67]:
tabulate(df["rating"])

Freq.  Perc.    Cum
1.0     20  0.002  0.002
1.2     23  0.003  0.005
1.3     14  0.002  0.006
1.5     10  0.001  0.008
1.6     24  0.003  0.010
1.7     12  0.001  0.012
1.8     10  0.001  0.013
2.0     54  0.006  0.019
2.2    165  0.019  0.038
2.4     82  0.009  0.047
2.5    242  0.028  0.075
2.7    295  0.034  0.108
2.9    148  0.017  0.125
3.0    515  0.059  0.184
3.2    607  0.069  0.253
3.4    306  0.035  0.288
3.5    731  0.083  0.371
3.7    837  0.095  0.467
3.9    393  0.045  0.512
4.0   1023  0.117  0.628
4.1    462  0.053  0.681
4.3    503  0.057  0.738
4.4    569  0.065  0.803
4.5    816  0.093  0.896
4.6    249  0.028  0.924
4.8    175  0.020  0.944
4.9    110  0.013  0.957
5.0     81  0.009  0.966
NaN    298  0.034  1.000

<p>Now that is a <i>great</i> function! Let's take not of a few observations:</p>
<ul>
    <li>The distribution of <code>"rating"</code> seems a littke skewed, with a longer left tail, maybe? We will figure this out when doing a EDA.</li>
    <li>We can affirm that most ratings are concentrated in the 3 - 4 stars interval</li>
    <li>By far, 4 stars is the most frequent observed value</li>
    <li>There are 298 <code>NaN</code> observations, which corresponds to 3.4% of the dataset.</li>
</ul>
<p>Let's now apply this same function to check <code>"rating_reviewcount"</code> and see the distribution of reviews:</p>

In [68]:
tabulate(df["rating_reviewcount"])

Freq.  Perc.    Cum
1.0        97  0.011  0.011
2.0        75  0.009  0.020
3.0        50  0.006  0.025
4.0        55  0.006  0.032
5.0         8  0.001  0.032
7.0        42  0.005  0.037
8.0        24  0.003  0.040
9.0        33  0.004  0.044
10.0       41  0.005  0.048
11.0       20  0.002  0.051
12.0        1  0.000  0.051
13.0       13  0.001  0.052
14.0       60  0.007  0.059
15.0       20  0.002  0.061
16.0       48  0.005  0.067
17.0       20  0.002  0.069
18.0        5  0.001  0.070
19.0       42  0.005  0.075
20.0       22  0.003  0.077
21.0       33  0.004  0.081
22.0       14  0.002  0.082
23.0       28  0.003  0.086
24.0       36  0.004  0.090
25.0       35  0.004  0.094
26.0       28  0.003  0.097
27.0       16  0.002  0.099
28.0       39  0.004  0.103
29.0       29  0.003  0.106
30.0       23  0.003  0.109
31.0       35  0.004  0.113
32.0       17  0.002  0.115
33.0       23  0.003  0.118
34.0       26  0.003  0.121
35.0       11  0.001  0.122
36.0       21  0.002  0.124
37.0       38  0.004  0.129
38.0       17  0.002  0.130
39.0       27  0.003  0.134
40.0       10  0.001  0.135
41.0       58  0.007  0.141
42.0       20  0.002  0.144
43.0       12  0.001  0.145
44.0       22  0.003  0.147
45.0       19  0.002  0.150
46.0       23  0.003  0.152
47.0       55  0.006  0.159
48.0       42  0.005  0.163
49.0       28  0.003  0.167
50.0       26  0.003  0.169
51.0       55  0.006  0.176
52.0       24  0.003  0.178
53.0       28  0.003  0.182
54.0       38  0.004  0.186
55.0       15  0.002  0.188
56.0       20  0.002  0.190
57.0       25  0.003  0.193
58.0       20  0.002  0.195
59.0       27  0.003  0.198
60.0       19  0.002  0.200
61.0       32  0.004  0.204
62.0        9  0.001  0.205
63.0       46  0.005  0.210
64.0       29  0.003  0.214
65.0       16  0.002  0.215
66.0       21  0.002  0.218
67.0        6  0.001  0.218
68.0       15  0.002  0.220
69.0       15  0.002  0.222
70.0       32  0.004  0.226
71.0       39  0.004  0.230
72.0        4  0.000  0.230
74.0       67  0.008  0.238
75.0       17  0.002  0.240
76.0       65  0.007  0.247
77.0       40  0.005  0.252
78.0       17  0.002  0.254
79.0       20  0.002  0.256
80.0       11  0.001  0.257
81.0       21  0.002  0.260
82.0       29  0.003  0.263
83.0        9  0.001  0.264
84.0       34  0.004  0.268
85.0        8  0.001  0.269
86.0       41  0.005  0.274
87.0       18  0.002  0.276
88.0       20  0.002  0.278
89.0       29  0.003  0.281
90.0       18  0.002  0.283
91.0       37  0.004  0.288
92.0       38  0.004  0.292
93.0       17  0.002  0.294
94.0       30  0.003  0.297
95.0       21  0.002  0.300
96.0       10  0.001  0.301
97.0       31  0.004  0.304
98.0       10  0.001  0.305
99.0       13  0.001  0.307
100.0       8  0.001  0.308
101.0      10  0.001  0.309
102.0      21  0.002  0.311
103.0      19  0.002  0.314
104.0       9  0.001  0.315
105.0      36  0.004  0.319
106.0       9  0.001  0.320
107.0      10  0.001  0.321
108.0      19  0.002  0.323
109.0      34  0.004  0.327
110.0      16  0.002  0.329
111.0       1  0.000  0.329
112.0      35  0.004  0.333
113.0      32  0.004  0.336
114.0       9  0.001  0.337
115.0      21  0.002  0.340
117.0      14  0.002  0.341
118.0      19  0.002  0.344
120.0      20  0.002  0.346
121.0      16  0.002  0.348
122.0       5  0.001  0.348
123.0      39  0.004  0.353
124.0       9  0.001  0.354
125.0      30  0.003  0.357
126.0      35  0.004  0.361
127.0      17  0.002  0.363
128.0       7  0.001  0.364
129.0      10  0.001  0.365
130.0      33  0.004  0.369
131.0      34  0.004  0.373
132.0      10  0.001  0.374
133.0      22  0.003  0.376
134.0      19  0.002  0.379
135.0      21  0.002  0.381
136.0       1  0.000  0.381
137.0       6  0.001  0.382
138.0      27  0.003  0.385
139.0      20  0.002  0.387
140.0      18  0.002  0.389
141.0      20  0.002  0.391
142.0      29  0.003  0.395
143.0      17  0.002  0.397
144.0      19  0.002  0.399
145.0      18  0.002  0.401
146.0      37  0.004  0.405
147.

<p>Now, most hotels seem to have a low number of review counts, and the distribution of this variable would probably have a long tail to the right. We would need to bin values to make a more accurate comment on it.</p>

In [69]:
df["rating_count"] = df["rating_reviewcount"].apply(float)
df["rating_count"].describe().to_frame().T

count        mean        std  min   25%    50%    75%     max
rating_count  8476.0  314.101227  408.48011  1.0  76.0  186.0  385.0  4300.0

<p>Now, let's rename the variables and assign more meaningful names:</p>

In [70]:
df = df.rename(
    columns={
        "rating2_ta": "ratingta",
        "rating2_ta_reviewcount": "ratingta_count",
        "addresscountryname": "country",
        "s_city": "city",
        "starrating": "stars",
    }
)

In [71]:
tabulate(df["stars"])

Freq.  Perc.    Cum
0.0     18  0.002  0.002
1.0      8  0.001  0.003
1.5     14  0.002  0.005
2.0    746  0.085  0.090
2.5    693  0.079  0.169
3.0   2527  0.288  0.457
3.5    692  0.079  0.535
4.0   2745  0.313  0.848
4.5    179  0.020  0.869
5.0   1152  0.131  1.000

<p>As noticed before, the distribution of stars is concentrated between the 3 - 4 interval with some gaps between stars. For instance, while there were 2745 observations for 4 stars (31%), there were only 179 for 4.5 stars (2%), while for 5 stars there were 1152 (13%).</p>
<p>We can now remove missing values with <code>dropping_missing</code> set as <code>True</code> in our <code>tabulate()</code> function and observe the distribution of values in <code>"rating"</code>.</p> 

In [72]:
tabulate(df["rating"], drop_missing=True)

Freq.  Perc.    Cum
1.0     20  0.002  0.002
1.2     23  0.003  0.005
1.3     14  0.002  0.007
1.5     10  0.001  0.008
1.6     24  0.003  0.011
1.7     12  0.001  0.012
1.8     10  0.001  0.013
2.0     54  0.006  0.020
2.2    165  0.019  0.039
2.4     82  0.010  0.049
2.5    242  0.029  0.077
2.7    295  0.035  0.112
2.9    148  0.017  0.130
3.0    515  0.061  0.190
3.2    607  0.072  0.262
3.4    306  0.036  0.298
3.5    731  0.086  0.384
3.7    837  0.099  0.483
3.9    393  0.046  0.529
4.0   1023  0.121  0.650
4.1    462  0.055  0.705
4.3    503  0.059  0.764
4.4    569  0.067  0.831
4.5    816  0.096  0.927
4.6    249  0.029  0.957
4.8    175  0.021  0.977
4.9    110  0.013  0.990
5.0     81  0.010  1.000

<p>We can remove unwanted columns:</p>

In [73]:
df = df.drop(
    columns=[
        "center2distance",
        "center1distance",
        "price_night",
        "guestreviewsrating",
        "rating_reviewcount",
    ]
)

In [74]:
df.head()

country city_actual center1label  \
52159  United Kingdom      London  City centre   
52160  United Kingdom      London  City centre   
52161  United Kingdom      London  City centre   
52162  United Kingdom      London  City centre   
52163  United Kingdom      London  City centre   

                             center2label neighbourhood  price    city  stars  \
52159  Wimbledon Park Underground Station    Blackheath    515  London    3.0   
52160  Wimbledon Park Underground Station    Blackheath    147  London    3.0   
52161  Wimbledon Park Underground Station    Blackheath    124  London    3.0   
52162  Wimbledon Park Underground Station    Blackheath    135  London    3.0   
52163  Wimbledon Park Underground Station    Blackheath    128  London    3.0   

       ratingta  ratingta_count  ...  year  month  weekend  holiday distance  \
52159       4.0          1669.0  ...  2017     12        0        1      6.5   
52160       4.0          1669.0  ...  2018      6        1        0      6.5   
52161       4.0          1669.0  ...  2018      3        1        0      6.5   
52162       4.0          1669.0  ...  2018      4        1        0      6.5   
52163       4.0          1669.0  ...  2018      2        1        0      6.5   

       distance_alter  accomodation_type  nnight  rating  rating_count  
52159             9.3              Hotel       4     4.0         161.0  
52160             9.3              Hotel       1     4.0         161.0  
52161             9.3              Hotel       1     4.0         161.0  
52162             9.3              Hotel       1     4.0         161.0  
52163             9.3              Hotel       1     4.0         161.0  

[5 rows x 25 columns]